Mise en place du Back-office

p3ters0ft opened the pull request:
- Utilisation du bundle Easy Admin
- l'accès au back-office sur via `/admin`
- Possibilité d'ajouter des utilisateurs via le BO
- Possibilité d'ajouter des entreprises et les branches

joshbradfield committed:
fix: Fix incompatibiliy with MySQL v5.7

The keyword 'VISIBILE' is not available in MySQL 5.7

This script was auto generated with help from MySQL Workbench,  To get the correct request I have reconfigured it to create requests for version 5.7

p3ters0ft committed:
Ajustement du flow connexion pour FO et BO

- Clone + Mis à jour du vendor
- Ajustement du flow connexion pour FO et BO

p3ters0ft committed:
Installation du Bundle Easy Admin

p3ters0ft committed:
Ajustement du code .htaccess

p3ters0ft committed:
Mis en place du Back-Office (provisoire)

- configuration du bundle Easy admin
- Premier menu [Liste des utilisateurs]

p3ters0ft committed:
[Entity] USER et COMPANY

Possibilité pour un utilisateur d'enregistrer 0 ou plusieurs entreprises

p3ters0ft committed:
[BO] Ajout de l'enttié COMPANY en back-office

sopalin committed:
Merge pull request #1 from Societes-AnitCOR/dev/feat_changements_table_company

fix: Fix incompatibiliy with MySQL v5.7

p3ters0ft committed:
Consigne les dates de création et modification de compte

p3ters0ft committed:
[Back-Office] Gestion des utilisateurs

joshbradfield commented:
### Quelle est le but des champs  `updated_at` and  `creation_at`

et quelle valeur par default ?

![image](https://user-images.githubusercontent.com/4537119/82242065-80bac500-993d-11ea-813d-7c469a76530c.png)

joshbradfield commented:
### Erreur sur la creation d'un user

1. ![image](https://user-images.githubusercontent.com/4537119/82242198-be1f5280-993d-11ea-8fb2-2fe21f98aff7.png)


2. Error   :

```
Catchable Fatal Error: Object of class App\Entity\Company could not be converted to string
```

​	url : [http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%252Fadmin%252F%253Fentity%253DUser%2526action%253Dlist%2526menuIndex%253D0%2526submenuIndex%253D-1%2526sortField%253Did%2526sortDirection%253DDESC%2526page%253D1](http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%2Fadmin%2F%3Fentity%3DUser%26action%3Dlist%26menuIndex%3D0%26submenuIndex%3D-1%26sortField%3Did%26sortDirection%3DDESC%26page%3D1)

![image](https://user-images.githubusercontent.com/4537119/82242180-b3fd5400-993d-11ea-8670-abe5b7c8fb30.png)

joshbradfield commented:
### Class de migration manquant 
utiliser la commande suivant pour générer la script de migration 
`php bin/console doctrine:migrations:generate`

p3ters0ft commented:
> ### Quelle est le but des champs `updated_at` and `creation_at`
> et quelle valeur par default ?
> 
> ![image](https://user-images.githubusercontent.com/4537119/82242065-80bac500-993d-11ea-813d-7c469a76530c.png)

Ce sont des champs de type integer. En fait, j'ai pensé qu'il serait intéressant de savoir quand est ce que un utilisateur aurait créé ou édité une entreprise.

Donc c'est un timestamp qui est prévu être renseigné

p3ters0ft commented:
> ### Class de migration manquant
> utiliser la commande suivant pour générer la script de migration
> `php bin/console doctrine:migrations:generate`

D'accord. Merci

p3ters0ft commented:
> ### Erreur sur la creation d'un user
> 1. ![image](https://user-images.githubusercontent.com/4537119/82242198-be1f5280-993d-11ea-8fb2-2fe21f98aff7.png)
> 2. Error   :
> 
> ```
> Catchable Fatal Error: Object of class App\Entity\Company could not be converted to string
> ```
> 
> ​ url : [http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%252Fadmin%252F%253Fentity%253DUser%2526action%253Dlist%2526menuIndex%253D0%2526submenuIndex%253D-1%2526sortField%253Did%2526sortDirection%253DDESC%2526page%253D1](http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%2Fadmin%2F%3Fentity%3DUser%26action%3Dlist%26menuIndex%3D0%26submenuIndex%3D-1%26sortField%3Did%26sortDirection%3DDESC%26page%3D1)
> 
> ![image](https://user-images.githubusercontent.com/4537119/82242180-b3fd5400-993d-11ea-8670-abe5b7c8fb30.png)

Corrigé  dans la branche `feat_easy_admin`

p3ters0ft committed:
[Back-Office] Correctif Ajout d'utilisateur + Gestion des coompagnies

joshbradfield commented:
> > ### Erreur sur la creation d'un user
> > 
> > 1. ![image](https://user-images.githubusercontent.com/4537119/82242198-be1f5280-993d-11ea-8fb2-2fe21f98aff7.png)
> > 2. Error   :
> > 
> > ```
> > Catchable Fatal Error: Object of class App\Entity\Company could not be converted to string
> > ```
> > 
> > 
> > ​ url : [http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%252Fadmin%252F%253Fentity%253DUser%2526action%253Dlist%2526menuIndex%253D0%2526submenuIndex%253D-1%2526sortField%253Did%2526sortDirection%253DDESC%2526page%253D1](http://localhost:8000/admin/?entity=User&action=new&menuIndex=0&submenuIndex=-1&sortField=id&sortDirection=DESC&page=1&referer=%2Fadmin%2F%3Fentity%3DUser%26action%3Dlist%26menuIndex%3D0%26submenuIndex%3D-1%26sortField%3Did%26sortDirection%3DDESC%26page%3D1)
> > ![image](https://user-images.githubusercontent.com/4537119/82242180-b3fd5400-993d-11ea-8670-abe5b7c8fb30.png)
> 
> Corrigé dans la branche `feat_easy_admin`


J'ai encore cette erreur :/

Je crois qu'il faut specifier un __toString() function pour l'entity "Company"
```php
class Company
{

    ...

    public function __toString(): string
    {
        return $this->name;
    }
}
```

joshbradfield committed:
dev : database migration for feat_easy_admin

p3ters0ft committed:
[BO]Wisiwig sur des champs en BO

Vous devez faire les actions suivantes:
- composer update
- php bin/console ckeditor:install
-  php bin/console assets:install --symlink

p3ters0ft committed:
Update .gitignore

p3ters0ft committed:
Merge branch 'feat_easy_admin_Migration' into feat_easy_admin

joshbradfield committed:
dev: Add timestamp to prevent null time values

joshbradfield committed:
Merge pull request #14 from Societes-AnitCOR/feat_easy_admin_Fixtures

fix: Add timestamp to prevent null time values

joshbradfield reviewed (approved):
Testé

Reinstallation et migration fonctionnalité
